tests/cross/src/test/scala/tests/pc/InlayHintsSuite.scala

```diff
@@ -870,7 +870,172 @@ class InlayHintsSuite extends BaseInlayHintsSuite {
+       |    println(s"Hello $string, $long, $integer!")
+       |  }
+       |  implicit def theString(implicit i: Int): String = i.toString
+       |  implicit def theInt: Int = 43
+       |  implicit def theLong: Long = 42
+       |  hello()
+       |}
+       |""".stripMargin,
+    """|object Main{
+       |  def hello()(implicit string: String, integer: Int, long: Long): String = {
+       |    println(s"Hello $string, $long, $integer!")
+       |  }
+       |  implicit def theString(implicit i: Int): String = i.toString
+       |  implicit def theInt: Int = 43
+       |  implicit def theLong: Long = 42
+       |  hello()/*(theString<<(5:15)>>(theInt<<(6:15)>>), theInt<<(6:15)>>, theLong<<(7:15)>>)*/
+       |}
+       |""".stripMargin,
+    compat = Map(
+      "3" -> """|object Main{
+                |  def hello()(implicit string: String, integer: Int, long: Long): String = {
+                |    println(s"Hello $string, $long, $integer!")
+                |  }
+                |  implicit def theString(implicit i: Int): String = i.toString
+                |  implicit def theInt: Int = 43
+                |  implicit def theLong: Long = 42
+                |  hello()/*(using theString<<(5:15)>>(theInt<<(6:15)>>), theInt<<(6:15)>>, theLong<<(7:15)>>)*/
+                |}
+                |""".stripMargin
+    )
+  )
+
+  check(
+    "implicit-fn",
+    """|object Main{
+       |  implicit def stringLength(s: String): Int = s.length
+       |  implicitly[String => Int]
+       |
+       |  implicit val namedStringLength: String => Long = (s: String) => s.length.toLong
+       |  implicitly[String => Long]
+       |}
+       |""".stripMargin,
+    """|object Main{
+       |  implicit def stringLength(s: String): Int = s.length
+       |  implicitly[String => Int]/*((s : String<<java/lang/String#>>) => stringLength<<(2:15)>>(s<<(3:12)>>)))*/
+       |
+       |  implicit val namedStringLength: String => Long = (s: String) => s.length.toLong
+       |  implicitly[String => Long]/*(namedStringLength<<(5:15)>>)*/
+       |}
+       |""".stripMargin,
+    compat = Map(
+      "3" -> """|object Main{
+                |  implicit def stringLength(s: String): Int = s.length
+                |  implicitly[String => Int]
+                |
+                |  implicit val namedStringLength: String => Long = (s: String) => s.length.toLong
+                |  implicitly[String => Long]/*(using namedStringLength<<(5:15)>>)*/
+                |}
+                |""".stripMargin
+    )
+  )
+
+  check(
+    "implicit-fn2",
+    """|object Main{
+       |  implicit def stringLength(s: String, i: Int): Int = s.length
+       |  implicitly[(String, Int) => Int]
+       |}
+       |""".stripMargin,
+    """|object Main{
+       |  implicit def stringLength(s: String, i: Int): Int = s.length
+       |  implicitly[(String, Int) => Int]/*((s : String<<java/lang/String#>>, i : Int<<scala/Int#>>) => stringLength<<(2:15)>>(s<<(3:12)>>, i<<(3:12)>>)))*/
```

Comment:
I fixed it to also attach information for go to definition to types here.

Reply:
I avoided this explicitly since I think the positions of the synthetic symbols are all offsets and just point to the initial implicit method application. It would be cool to be able to calculate positions inside of inlays somehow.

Reply:
Right, but the positions of those symbols were actually present, I added the `<<scala/Int#>>` part, which will correctly resolve to the type.

We can remove the positions from the synthetic symbols, shouldn't be an issue

Reply:
Sorry! I see now what you meant, actually I Had some code to remove positions from synthetic symbols, but it broke other tests, so I removed it.

Reply:
Ok, I fixed that to not show those positions of synthetic params